lib/scaffold.dart

```diff
@@ -574,6 +615,10 @@ class BackdropScaffoldState extends State<BackdropScaffold>
                 fit: StackFit.expand,
                 children: <Widget>[
                   _buildBackPanel(),
+                  if (isBackLayerConcealed && widget.frontLayerActiveFactor < 1)
+                    Container(
+                        color: _backLayerScrimColorTween.evaluate(controller),
+                        height: _backPanelHeight),
```

Comment:
```suggestion
                  if (isBackLayerConcealed && widget.frontLayerActiveFactor < 1)
                    Container(
                        color: _backLayerScrimColorTween.evaluate(controller),
                        height: _backPanelHeight),
```

Can we put this inside `_buildBackPanel`?

Reply:
Here it's the middle element of a `Stack` -- back layer, conditional scrim, front layer.  

Instead of making another Stack inside `_buildBackPanel`, what do you think about just extracting the scrim?

```suggestion
                  if (_hasBackLayerScrim) _buildBackLayerScrim(),
                  ...
              );
          }
  
  // Note: I would prefer "extract widget" refactoring to "extract builder method," 
  // but this keeps it consistent
  Container _buildBackLayerScrim() => Container(
      color: _backLayerScrimColorTween.evaluate(controller),
      height: _backPanelHeight);

  bool get _hasBackLayerScrim =>
      isBackLayerConcealed && widget.frontLayerActiveFactor < 1;
```

Reply:
> It looks like a mistake, i dont know how it landed! Did you test the fix properly? It isnt breaking anything else right?

It seems OK.

Reply:
the "scrims part" are kept in their indivdual layer, to avoid "scrim layer" escaping either "frontLayer" or "backLayer", by mistake. This why it is also done in same way for "front layer". Also "sematically" scrim in part of the layer!

Although you could extract of "scrim layer" but should be used inside `_buildFrontPlanel` or `_buildBackPanel`.

Reply:
`_buildBackPanel` now returns a `Stack` with the scrim on top.

Reply:
👍